apps/pipelines/nodes/nodes.py

```diff
@@ -96,25 +98,88 @@ def _process(self, state: PipelineState) -> PipelineState:
         return input
 
 
-class ExtractStructuredDataBasic(LLMResponse):
-    __human_name__ = "Extract structured data (Basic)"
-    data_schema: str
-
-    def _process(self, state: PipelineState) -> RunnableLambda:
-        json_schema = ExtractStructuredDataBasic.to_json_schema(json.loads(self.data_schema))
-        prompt = PromptTemplate.from_template(template="{input}.\nCurrent user data: {participant_data}")
-        chain = (
+class ExtractStructuredDataNodeMixin:
+    def _prompt_chain(self, reference_data):
+        prompt = PromptTemplate.from_template(template="Current user data: {reference_data}\nConversations: {input}")
```

Comment:
Is this the full prompt? Should it be expanded a bit to include more instruction?

Reply:
Sure. It seems to be doing great as-is, but that may just be due to it using OpenAI's json data endpoint, which other providers would not necessarily have